src/api_server/src/request/actions.rs

```diff
@@ -30,27 +29,8 @@ struct ActionBody {
 
 fn validate_payload(action_body: &ActionBody) -> Result<(), Error> {
     match action_body.action_type {
-        ActionType::BlockDeviceRescan => {
-            match action_body.payload {
-                Some(ref payload) => {
-                    // Expecting to have drive_id as a String in the payload.
-                    if !payload.is_string() {
-                        return Err(Error::Generic(
-                            StatusCode::BadRequest,
-                            "Invalid payload type. Expected a string representing the drive_id"
-                                .to_string(),
-                        ));
-                    }
-                    Ok(())
-                }
-                None => Err(Error::Generic(
-                    StatusCode::BadRequest,
-                    "Payload is required for block device rescan.".to_string(),
-                )),
-            }
-        }
         ActionType::FlushMetrics | ActionType::InstanceStart | ActionType::SendCtrlAltDel => {
-            // Neither FlushMetrics nor InstanceStart should have a payload.
+            // Neither of these actions should have a payload.
             if action_body.payload.is_some() {
```

Comment:
Please remove `payload` field altogether from the `ActionBody` struct. It was only used with `BlockDeviceRescan` and was complicating the code because of all the necessary checking for/against its presence. Let's take this opportunity to clean that up as well.

Reply:
As it would require more changes to the API to get rid of `ActionBody` completely, I only removed the `payload` field.